### PR TITLE
[Bug] Cleanup test_event_override unit test

### PR DESCRIPTION
**Pull request by terrancedejesus**
## Issues
* https://github.com/elastic/detection-rules/issues/3394

## Summary
This PR fixes the `TestRuleTiming.test_event_override` unit test by doing the following:
- Removes `rule_integrations` variable as it is not used in the logic to determine if the `timestamp_override` field is necessary
- Removes `rule_query` since we are using `is_sequence()` method to determine if the EQL query uses sequences
- Changes `"sequence" in rule_query` to `rule.contents.data.is_sequence`

Note: While `is_sequence` is a method of `EQLRuleData`, it exists as an attribute of this object because `QueryRuleData` is inherited into these dataclass. We only run the rest of this unit test if the `rule.contents.data` is `QueryRuleData`.

### CODE REVIEW

File: tests/test_all_rules.py [hunk 1]
```diff
@@ -964,18 +964,14 @@ def test_event_override(self):
...
-                rule_query = rule.contents.data.get('query')
                 has_event_ingested = rule.contents.data.get('timestamp_override') == 'event.ingested'
                 rule_str = self.rule_str(rule, trailer=None)
 
                 if not has_event_ingested:
                     # TODO: determine if we expand this to ES|QL
                     # ignores any rule that does not use EQL or KQL queries specifically
                     # this does not avoid rule types where variants of KQL are used (e.g. new terms)
-                    if rule_language not in ('eql', 'kuery') or "sequence" in rule_query:
+                    if rule_language not in ('eql', 'kuery') or rule.contents.data.is_sequence:
```
Comment (Mikaayenson) [inline tests/test_all_rules.py:974]: Now that I'm seeing this, when do we actually run into this case where we have an a language not eql but still is a sequence?

Comment (terrancedejesus) [reply tests/test_all_rules.py:974]: Never, sequences are only in EQL.

Comment (Mikaayenson) [reply tests/test_all_rules.py:974]: so do we even need the is_sequence check at all?

**Review by Mikaayenson — APPROVED** [review]
LGTM!

**Review by shashank-elastic — APPROVED** [review]

**Review by eric-forte-elastic — APPROVED** [review]
:green_circle: Manual review, looks good to me! :+1:

**Pull request merged**